gridmenuview/src/main/java/com/bcc/gridmenuview/adapter/GridMenuAdapter.java

```diff
@@ -40,11 +42,24 @@ public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
     @Override
     public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
         holder.title.setText(menuItems.get(position).getTitle());
-        holder.image.setImageDrawable(menuItems.get(position).getImage());
+        if (menuItems.get(position).getUrl() == null) {
+            holder.image.setImageDrawable(menuItems.get(position).getImage());
+        } else {
+            ImageUrlAsyncTask imageAsync = (ImageUrlAsyncTask) new ImageUrlAsyncTask()
+                    .execute(menuItems.get(position).getUrl(), holder.image.getContext());
+            try {
+                holder.image.setImageDrawable(imageAsync.get());
+            } catch (ExecutionException e) {
+                e.printStackTrace();
+            } catch (InterruptedException e) {
```

Comment:
is it required to have double catch here?

Reply:
> is it required to have double catch here?

I don't think so. It can also only use catch exception

Reply:
yeepp, or you can create a single class to handle all the exceptions

Reply:
> yeepp, or you can create a single class to handle all the exceptions

Okayy, Thank's for the advice